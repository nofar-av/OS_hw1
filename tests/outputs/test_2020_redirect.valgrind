==26842== Memcheck, a memory error detector
==26842== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==26842== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==26842== Command: /home/omer/OS_wet1/smash
==26842== Parent PID: 26840
==26842== 
==26842== 
==26842== FILE DESCRIPTORS: 19 open at exit.
==26842== Open file descriptor 18: sl
==26842==    at 0x54EEC3E: open (open64.c:47)
==26842==    by 0x110A4F: RedirectionCommand::execute() (Commands.cpp:417)
==26842==    by 0x11A242: SmallShell::executeCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) (small_shell.cpp:171)
==26842==    by 0x118BCD: main (smash.cpp:42)
==26842== 
==26842== Open file descriptor 17: test/again
==26842==    at 0x54EEC3E: open (open64.c:47)
==26842==    by 0x110A4F: RedirectionCommand::execute() (Commands.cpp:417)
==26842==    by 0x11A242: SmallShell::executeCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) (small_shell.cpp:171)
==26842==    by 0x118BCD: main (smash.cpp:42)
==26842== 
==26842== Open file descriptor 16: new
==26842==    at 0x54EEC3E: open (open64.c:47)
==26842==    by 0x110A4F: RedirectionCommand::execute() (Commands.cpp:417)
==26842==    by 0x11A242: SmallShell::executeCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) (small_shell.cpp:171)
==26842==    by 0x118BCD: main (smash.cpp:42)
==26842== 
==26842== Open file descriptor 15: new
==26842==    at 0x54EEC3E: open (open64.c:47)
==26842==    by 0x110A4F: RedirectionCommand::execute() (Commands.cpp:417)
==26842==    by 0x11A242: SmallShell::executeCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) (small_shell.cpp:171)
==26842==    by 0x118BCD: main (smash.cpp:42)
==26842== 
==26842== Open file descriptor 14: new
==26842==    at 0x54EEC3E: open (open64.c:47)
==26842==    by 0x110A4F: RedirectionCommand::execute() (Commands.cpp:417)
==26842==    by 0x11A242: SmallShell::executeCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) (small_shell.cpp:171)
==26842==    by 0x118BCD: main (smash.cpp:42)
==26842== 
==26842== Open file descriptor 13: p
==26842==    at 0x54EEC3E: open (open64.c:47)
==26842==    by 0x110A4F: RedirectionCommand::execute() (Commands.cpp:417)
==26842==    by 0x11A242: SmallShell::executeCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) (small_shell.cpp:171)
==26842==    by 0x118BCD: main (smash.cpp:42)
==26842== 
==26842== Open file descriptor 12: t
==26842==    at 0x54EEC3E: open (open64.c:47)
==26842==    by 0x110A4F: RedirectionCommand::execute() (Commands.cpp:417)
==26842==    by 0x11A242: SmallShell::executeCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) (small_shell.cpp:171)
==26842==    by 0x118BCD: main (smash.cpp:42)
==26842== 
==26842== Open file descriptor 11: k
==26842==    at 0x54EEC3E: open (open64.c:47)
==26842==    by 0x110A4F: RedirectionCommand::execute() (Commands.cpp:417)
==26842==    by 0x11A242: SmallShell::executeCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) (small_shell.cpp:171)
==26842==    by 0x118BCD: main (smash.cpp:42)
==26842== 
==26842== Open file descriptor 10: s
==26842==    at 0x54EEC3E: open (open64.c:47)
==26842==    by 0x110A4F: RedirectionCommand::execute() (Commands.cpp:417)
==26842==    by 0x11A242: SmallShell::executeCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) (small_shell.cpp:171)
==26842==    by 0x118BCD: main (smash.cpp:42)
==26842== 
==26842== Open file descriptor 9: mypid
==26842==    at 0x54EEC3E: open (open64.c:47)
==26842==    by 0x110A4F: RedirectionCommand::execute() (Commands.cpp:417)
==26842==    by 0x11A242: SmallShell::executeCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) (small_shell.cpp:171)
==26842==    by 0x118BCD: main (smash.cpp:42)
==26842== 
==26842== Open file descriptor 8: mypid
==26842==    at 0x54EEC3E: open (open64.c:47)
==26842==    by 0x1109F3: RedirectionCommand::execute() (Commands.cpp:409)
==26842==    by 0x11A242: SmallShell::executeCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) (small_shell.cpp:171)
==26842==    by 0x118BCD: main (smash.cpp:42)
==26842== 
==26842== Open file descriptor 7: mypid
==26842==    at 0x54EEC3E: open (open64.c:47)
==26842==    by 0x110A4F: RedirectionCommand::execute() (Commands.cpp:417)
==26842==    by 0x11A242: SmallShell::executeCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) (small_shell.cpp:171)
==26842==    by 0x118BCD: main (smash.cpp:42)
==26842== 
==26842== Open file descriptor 6: new
==26842==    at 0x54EEC3E: open (open64.c:47)
==26842==    by 0x1109F3: RedirectionCommand::execute() (Commands.cpp:409)
==26842==    by 0x11A242: SmallShell::executeCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) (small_shell.cpp:171)
==26842==    by 0x118BCD: main (smash.cpp:42)
==26842== 
==26842== Open file descriptor 5: new
==26842==    at 0x54EEC3E: open (open64.c:47)
==26842==    by 0x110A4F: RedirectionCommand::execute() (Commands.cpp:417)
==26842==    by 0x11A242: SmallShell::executeCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) (small_shell.cpp:171)
==26842==    by 0x118BCD: main (smash.cpp:42)
==26842== 
==26842== Open file descriptor 4: new
==26842==    at 0x54EEC3E: open (open64.c:47)
==26842==    by 0x110A4F: RedirectionCommand::execute() (Commands.cpp:417)
==26842==    by 0x11A242: SmallShell::executeCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) (small_shell.cpp:171)
==26842==    by 0x118BCD: main (smash.cpp:42)
==26842== 
==26842== Open file descriptor 3: /home/omer/OS_wet1/tests/outputs/test_2020_redirect.valgrind
==26842==    <inherited from parent>
==26842== 
==26842== Open file descriptor 2: /home/omer/OS_wet1/tests/outputs/test_2020_redirect.err
==26842==    <inherited from parent>
==26842== 
==26842== Open file descriptor 1: /home/omer/OS_wet1/tests/outputs/test_2020_redirect.out
==26842==    <inherited from parent>
==26842== 
==26842== Open file descriptor 0:
==26842==    <inherited from parent>
==26842== 
==26842== 
==26842== HEAP SUMMARY:
==26842==     in use at exit: 0 bytes in 0 blocks
==26842==   total heap usage: 359 allocs, 359 frees, 100,361 bytes allocated
==26842== 
==26842== All heap blocks were freed -- no leaks are possible
==26842== 
==26842== For counts of detected and suppressed errors, rerun with: -v
==26842== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
