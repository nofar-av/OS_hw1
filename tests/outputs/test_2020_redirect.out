smash> smash> smash> smash> smash> test_redirect
smash> smash> again_redirect
smash> smash> again_redirect
test_redirect
smash> smash> smash pid is 5254
smash> smash> smash pid is 5254
smash pid is 5254
smash> smash> smash pid is 5254
smash> mypid
new
smash> smash> smash> [1] sleep 100& : 5317 0 secs
smash> smash> smash> smash> smash> k
mypid
new
s
t
smash> smash> k
mypid
new
p
s
t
smash> k
mypid
new
p
s
t
smash> smash> smash> before
smash> again_redirect
test_redirect
smash> smash> smash> smash> k
mypid
p
s
t
smash> smash> k
mypid
new
p
s
t
smash> k
mypid
new
p
s
t
smash> smash> smash> k
mypid
new
p
s
t
smash> k
mypid
new
p
s
t
smash> smash> smash> k
mypid
new
p
s
t
smash> smash> smash> k
mypid
new
p
s
t
test
smash> smash> 0
2
end: 4
smash> smash> smash> smash: sending SIGKILL signal to 1 jobs:
5317: sleep 100&
